pil-analyzer/src/evaluator.rs

```diff
@@ -527,6 +527,7 @@ pub trait SymbolLookup<'a, T> {
     }
 
     fn eval_expr(&self, _expr: &AlgebraicExpression<T>) -> Result<Arc<Value<'a, T>>, EvalError> {
+        // TODO: Is this correct?
```

Comment:
This is just the default implementation of the trait. The query processor should supply a different one that actually performs the evaluation.

Reply:
Yeah, but the query processor is not used, but instead the `Definitions` struct, see my changes in there. This is hacky of course, need to understand the different implementations of `SymbolLookup` a bit better! Perhaps the default implementation could already handle literals?